pkgs/misc/mox-boot-builder/default.nix

```diff
@@ -0,0 +1,46 @@
+{ stdenv
+, lib
+, fetchgit
+, buildPackages
+, pkgsCross
+}:
+
+stdenv.mkDerivation rec {
+  pname = "mox-boot-builder";
```

Comment:
This name is inaccurate. while this repo is named `mox-boot-builder`, what it actually builds is a firmware that adds support for the hardware random number generator.

 - https://github.com/ARM-software/arm-trusted-firmware/blob/5a53c6c6679e189eda7233232570958263332883/docs/plat/marvell/armada/build.rst?plain=1#L273-L295

I'm not sure what really it should be named. `armada-3720-secure-firmware` wouldn't be bad, but that name would *also* be somewhat wrong, since it's more like `nic-cz-armada-3720-hardware-rng-firmware`. Though that would be a mouthful.

Anyone has suggestions?

Reply:
As far as I am aware, there are two versions of BL32 software for armada 3700 (a3k) chips: the initial marvell code which does nothing more than programs some efuses, and the turris code which provides the hwrng.  Turris can sign their firmware for their Mox boards; here we use it unsigned.  I have offered "nic-cz-armada-bl32-unsigned-firmware".